src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1377 @@
+            srcCount = BuildPutArgReg(tree->AsUnOp());
+            break;
+
+        case GT_CALL:
+            srcCount = BuildCall(tree->AsCall());
+            if (tree->AsCall()->HasMultiRegRetVal())
+            {
+                dstCount = tree->AsCall()->GetReturnTypeDesc()->GetReturnRegCount();
+            }
+            break;
+
+        case GT_ADDR:
+        {
+            // For a GT_ADDR, the child node should not be evaluated into a register
+            GenTree* child = tree->gtGetOp1();
+            assert(!isCandidateLocalRef(child));
+            assert(child->isContained());
+            assert(dstCount == 1);
+            srcCount = 0;
+            BuildDef(tree);
+        }
+        break;
+
+        case GT_BLK:
+            // These should all be eliminated prior to Lowering.
+            assert(!"Non-store block node in Lowering");
+            srcCount = 0;
+            break;
+
+        case GT_STORE_BLK:
+        case GT_STORE_OBJ:
+        case GT_STORE_DYN_BLK:
+            srcCount = BuildBlockStore(tree->AsBlk());
+            break;
+
+        case GT_INIT_VAL:
+            // Always a passthrough of its child's value.
+            assert(!"INIT_VAL should always be contained");
+            srcCount = 0;
+            break;
+
+        case GT_LCLHEAP:
+        {
+            assert(dstCount == 1);
+
+            // Need a variable number of temp regs (see genLclHeap() in codegenloongarch64.cpp):
+            // Here '-' means don't care.
+            //
+            //  Size?                   Init Memory?    # temp regs
+            //   0                          -               0
+            //   const and <=UnrollLimit    -               0
+            //   const and <PageSize        No              0
+            //   >UnrollLimit               Yes             0
+            //   Non-const                  Yes             0
+            //   Non-const                  No              2
+            //
+
+            GenTree* size = tree->gtGetOp1();
+            if (size->IsCnsIntOrI())
+            {
+                assert(size->isContained());
+                srcCount = 0;
+
+                size_t sizeVal = size->AsIntCon()->gtIconVal;
+
+                if (sizeVal != 0)
+                {
+                    // Compute the amount of memory to properly STACK_ALIGN.
+                    // Note: The Gentree node is not updated here as it is cheap to recompute stack aligned size.
+                    // This should also help in debugging as we can examine the original size specified with
+                    // localloc.
+                    sizeVal         = AlignUp(sizeVal, STACK_ALIGN);
+                    size_t insCount = sizeVal / (REGSIZE_BYTES * 2);
+
+                    // For small allocations up to 4 'st' instructions (i.e. 16 to 64 bytes of localloc)
+                    // TODO-LoongArch64: maybe use paird-load/store or SIMD in future.
+                    if (insCount <= 4)
```

Comment:
Why not just change to `sizeVal <= (REGSIZE_BYTES * 2 * 4)`?

Reply:
Thanks